librz/analysis/esil_trace.c

```diff
@@ -53,7 +53,9 @@ RZ_API RzAnalysisEsilTrace *rz_analysis_esil_trace_new(RzAnalysisEsil *esil) {
 		if (!b) {
 			goto error;
 		}
-		memcpy(b->bytes, a->bytes, b->size);
+		if (b->bytes && a->bytes && b->size > 0) {
```

Comment:
this check is IMHO wrong. You should not do such checks, because I think RzReg should have an invariant that says that `->bytes` is always allocated. In `rz_reg_arena_new` you see that if the allocation `->bytes` fails, the whole RzRegArena is not created.

Reply:
Why is this necessary? If it really is, I suspect the real problem is somewhere else.

Reply:
It is necessary because often `rz_reg_arena()` created with 0 size during the profile loading. Rizin calls `rz_reg_profile_to_cc()` multiple times during the RzBin loading process, when `asm.arch` and `asm.bits` not propagated yet. I know, it's a mess, but proper fix would be a RzBin refactoring loading process.

Reply:
If you return NULL in that case, the whole bin loading is broken :D

Reply:
What does `rz_reg_profile_to_cc`  has to do with RzRegArena?

Reply:
@ret2libc it's a long chain of calls, setting up the calling convention during the binary loading, which then takes `asm.arch` and `asm.bits` to load default regprofile for that architecture to produce the calling convention list, etc.

Reply:
I still fail to see where `->bytes` is NULL and `->size != 0`. All methods in `reg/arena.c` that allocate `->bytes` also check if it is NULL and if it is they usually return false/NULL and free the arena. There are only a bunch of cases where `->bytes` can become NULL:
1) in `rz_reg_fit_arena`, but in that case also `->size` is set to 0, so the memcpy should not do anything.
2) in `rz_reg_arena_new` in case the size is 0, also `->bytes` is set to NULL, but then again the memcpy should not do anything.

Reply:
Ok, but too bad! Eventually, I think we don't want these kind of checks. If a RzRegArena exists, ->bytes should never be null so to avoid all these problems and stay consistent.

Reply:
Of course, but for this the function should be called only when everything is ready, not multiple times during the RzBin loading process. One day...